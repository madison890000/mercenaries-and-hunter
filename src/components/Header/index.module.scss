.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
}


.basicInfo {
  text-align: right;
}

.email {
  font-style: italic;
  font-size: 16px;
}

.cellphone {
  font-style: italic;
  font-size: 16px;
}
