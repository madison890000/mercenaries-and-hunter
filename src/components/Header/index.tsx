import React from 'react';
import styles from './index.module.scss';
import capitalize from '../../utils/capitalize';
import DataModel from '../../models/types';

interface HeaderProps {
    firstName: string;
    lastName: string;
    email: string;
    cellphone: string;
    links?: DataModel.RelatedLink[];
}

const Header = ({ links, firstName, lastName, email, cellphone }: HeaderProps) => {
    return (
        <header className={styles.header}>
            <h1>
                {capitalize(firstName)} {capitalize(lastName)}
            </h1>
            <div className={styles.basicInfo}>
                <div className={styles.basicInfoItem}>
                    <div>Email:</div>
                    <div>{email}</div>
                </div>
                <div className={styles.basicInfoItem}>
                    <div>Cellphone:</div>
                    <div>{cellphone}</div>
                </div>
                <>
                    {links?.map(link => (
                        <div className={styles.basicInfoItem}>
                            <div>{link?.name}:</div>
                            <div>
                                <a href={link?.value} target="_blank">
                                    {link?.value}
                                </a>
                            </div>
                        </div>
                    ))}
                </>
            </div>
        </header>
    );
};

export default Header;
