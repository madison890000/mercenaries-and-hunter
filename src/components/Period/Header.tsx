import Tag from '../Tag';
import React from 'react';
import DataModel from '../../models/types';
import styles from './Header.module.scss';
import StringWithID from '../../models/StringWithID';
import capitalize from '../../utils/capitalize';

interface HeaderProps {
    keywords: StringWithID[];
    jobPositionLevel: DataModel.JobPositionLevel;
    jobPosition: DataModel.JobPosition;
    companyName: string;
    companyType: DataModel.CompanyType;
}

const Header = ({ keywords, companyName, jobPosition, jobPositionLevel }: HeaderProps) => (
    <div className={styles.header}>
        <span className={styles.job}>
            {jobPositionLevel && <span>{capitalize(jobPositionLevel.toLowerCase())}</span>}
            <span>{jobPosition}</span>
        </span>
        <span>
            {keywords?.map(keyword => (
                <Tag type="filled" key={keyword.id}>
                    {keyword.toString()}
                </Tag>
            ))}
        </span>
        <span className={styles.companyName}>
            <span>{companyName}</span>
        </span>
    </div>
);

export default Header;
