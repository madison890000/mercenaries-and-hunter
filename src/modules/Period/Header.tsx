import Tag from '../../components/Tag';
import React from 'react';
import DataModel from '../../models/types';
import styles from './Header.module.scss';
import capitalize from '../../utils/capitalize';
import Divider from '../../components/Divider';

interface HeaderProps {
    keywords: string[];
    jobPositionLevel: DataModel.JobPositionLevel;
    jobPosition: DataModel.JobPosition;
    companyName: string;
    companyIndustry: string;
    companyType: DataModel.CompanyType;
}

const Header = ({keywords, companyName, companyIndustry, jobPosition, jobPositionLevel}: HeaderProps) => (
    <div className={styles.header}>
        <div>
            <div className={styles.job}>
                {jobPositionLevel && <span>{capitalize(jobPositionLevel.toLowerCase())}</span>}
                <span>{jobPosition}</span>
            </div>
            <div className={styles.companyName}>
                <div>{companyName}</div>
                {/*<div className={styles.companyIndustry}>({capitalize(companyIndustry)})</div>*/}
            </div>
        </div>
        <Divider variant="v"/>
        <div style={{
            fontSize: 24
        }}>
            {keywords?.map(keyword => (
                <Tag type="filled" key={keyword}>
                    {keyword.toString()}
                </Tag>
            ))}
        </div>
    </div>
);

export default Header;
