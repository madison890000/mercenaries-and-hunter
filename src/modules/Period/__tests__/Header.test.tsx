import React from 'react';
import Header from '../Header';
import renderer from 'react-test-renderer';
import DataModel from '../../../models/types';

test('renders Period Header', () => {
    const testCompanyName = 'test company name';
    const keywords = ['React'];
    const jobPositionLevel = DataModel.JobPositionLevel.Senior;
    const jobPosition = 'Software Engineer';
    const HeaderDom = renderer
        .create(
            <Header
                companyType={DataModel.CompanyType.Startup}
                jobPosition={jobPosition}
                jobPositionLevel={jobPositionLevel}
                companyName={testCompanyName}
                keywords={keywords}
                companyIndustry={'tele'}
            />
        )
        .toJSON();
    expect(HeaderDom).toMatchSnapshot();
});
