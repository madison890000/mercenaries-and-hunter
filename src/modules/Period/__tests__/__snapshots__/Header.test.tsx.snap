// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Period Header 1`] = `
<div
  className="header"
>
  <div>
    <div
      className="job"
    >
      <span>
        Senior
      </span>
      <span>
        Software Engineer
      </span>
    </div>
    <div
      className="companyName"
    >
      <div>
        test company name
      </div>
    </div>
  </div>
  <div
    className="dividerH"
  />
  <div
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    <div
      className="filled tag"
    >
      React
    </div>
  </div>
</div>
`;
